cli/run_config.go

```diff
@@ -0,0 +1,57 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/bitrise-io/bitrise/configs"
+	envmanModels "github.com/bitrise-io/envman/models"
+	"github.com/bitrise-io/go-utils/colorstring"
+	log "github.com/sirupsen/logrus"
+)
+
+func getNoOutputTimoutValue(inventoryEnvironments []envmanModels.EnvironmentItemModel) (string, error) {
+	for _, env := range inventoryEnvironments {
+		key, value, err := env.GetKeyValuePair()
+		if err != nil {
+			return "", err
+		}
+
+		if key == configs.NoOutputTimeoutEnvKey && value != "" {
+			return value, nil
+		}
+	}
+
+	return os.Getenv(configs.NoOutputTimeoutEnvKey), nil
+}
+
+func readNoOutputTimoutConfiguration(inventoryEnvironments []envmanModels.EnvironmentItemModel) time.Duration {
+	const defaultTimeout = 0
```

Comment:
I think it would be a bit safer to use `-1` as the off value here instead of `0`. A zero duration is something that other parts of the codebase can still interpret as a timeout and activate the hang detector accidentally.